charts/jaeger-v2/readme.md

```diff
@@ -0,0 +1,167 @@
+# Jaeger Helm Chart Deployment (Version 2) Documentation
+
+## Overview
+
+This document outlines the steps to deploy the Jaeger Helm chart (version 2) and provides information on how to use a custom `values.yaml` configuration file during installation. The Jaeger Helm chart includes multiple sub-charts for data stores like Cassandra, Elasticsearch, and Kafka.
+
+## Directory Structure
+
+The structure of the Jaeger Helm chart is as follows:
+
+```bash
+kali@PC:~/LFX/helm-charts/charts/jaeger-v2$ tree
+.
+├── Chart.lock
+├── Chart.yaml
+├── charts
+│   ├── cassandra-0.15.3.tgz
+│   ├── common-2.23.0.tgz
+│   ├── elasticsearch-20.0.4.tgz
+│   └── kafka-26.6.2.tgz
+├── readme.md
+├── templates
+│   ├── _helpers.tpl
+│   ├── config-map.yaml
+│   └── deployment.yaml
+└── values.yaml
+
+3 directories, 11 files
+```
+
+### Key Files
+- **Chart.yaml**: Contains metadata about the Helm chart such as the chart version, name, and dependencies.
+- **templates/**: This directory contains Kubernetes manifests that are templated by Helm and used for the deployment of the Jaeger application. Key templates include:
+  - `config-map.yaml`: Configures the ConfigMap for the application.
+  - `deployment.yaml`: Contains the template for the Kubernetes deployment of Jaeger.
+- **values.yaml**: This file contains default configuration values for the chart, which are applied unless overridden by a custom file.
+- **charts/**: Contains Helm chart dependencies such as Cassandra, Elasticsearch, Kafka, and common utilities.
+- **readme.md**: Provides chart-specific documentation.
+
+## Helm Installation
+
+You can install the Jaeger Helm chart with the following commands:
+
+### Basic Installation
+
+To install the Jaeger chart using the default values from the included `values.yaml`:
+
+```bash
+helm install <chart_name> ./
+```
+
+- `<chart_name>` is the name you want to give your release.
+
+### Installation with Custom Values
+
+If you want to use a custom configuration, you can pass a custom `values.yaml` file. This allows you to override the default values specified in the chart’s `values.yaml`.
+
+Use the following command to install the chart with a custom values file:
+
+```bash
+helm install <chart_name> ./ -f <custom_values.yaml>
+```
+
+- Replace `<chart_name>` with the desired name for your Helm release.
+- Replace `<custom_values.yaml>` with the path to your custom configuration file.
+
+### Example:
+
+```bash
+helm install jaeger-v2 ./ -f my-config-values.yaml
+```
+
+This command installs the Jaeger chart and uses `my-config-values.yaml` for overriding any configuration specified in the default `values.yaml`.
```

Comment:
what does "overriding" mean? Does it completely ignore the default file, or performs some sort of merge?

Reply:
let's say you have specified
 ```
 jaeger_storage:
    backends:
      some_store:
        memory:
          max_traces: 90000
      another_store:
        memory:
          max_traces: 90000
 ````
 but didn't specified 
 ```
 remote_sampling:
    adaptive:
      sampling_store: some_store
      initial_sampling_probability: 0.1
 ```
 so it will use default value of 0.1 from `vaules.yaml`